crates/ruff/src/rules/flake8_simplify/snapshots/ruff__rules__flake8_simplify__tests__SIM101_SIM101.py.snap.new

```diff
@@ -0,0 +1,126 @@
+---
+source: crates/ruff/src/rules/flake8_simplify/mod.rs
+assertion_line: 49
+expression: diagnostics
+---
+- kind:
+    body: "Multiple `isinstance` calls for `a`, merge into a single call"
+    fixable: true
+    commit: "Merge `isinstance` calls for `a`"
+    rule: DuplicateIsinstanceCall
```

Comment:
I don't know if there's a way for us to avoid denormalizing the rule here. Previously, this would've been a `DuplicateIsinstanceCall` struct with its fields, rather than the materialized message, which is arguably better than what we have here.

Reply:
More succinctly, there isn't a way to go from `DiagnosticKind` back to the originating `Violation` struct. Instead, you can only go from `DiagnosticKind` to `Rule` (which is a singleton object, not something that contains the relevant fields).